### PR TITLE
Make token semantics match actions/checkout

### PR DESCRIPTION
**Pull request by jcbhmr**
this pr would...
- [x] make it say `token: ${{ secrets.TOKENHERE }}` in readme
- [x] use `https://x:$TOKEN@github.com/user/repo.wiki.git` in git ops
- [x] set default to `token: ${{ github.token }}` in action.yml
- [x] not change anything else

### CODE REVIEW

**Comment by jcbhmr** [comment]
@spenserblack request review

File: src/index.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -e
 # shellcheck source=src/git_exists.sh
 source "$(dirname "${BASH_SOURCE[0]}")/git_exists.sh"
 
-if ! git_exists "https://$INPUT_TOKEN@$INPUT_REPOSITORY.wiki.git" &>/dev/null; then
+if ! git_exists "https://x:$INPUT_TOKEN@$INPUT_REPOSITORY.wiki.git" &>/dev/null; then
```
Comment (spenserblack) [inline src/index.sh:7]: Even if this value is useless, I'd like it to default to the current user or something. However, before addressing this, check my other comment.

**Review by spenserblack — COMMENTED** [review]
I just remembered that `gh` is installed in actions, and we can use that.
Just set the environment variable `GH_TOKEN` when using `gh`, and clone with `gh repo clone ${{ inputs.repository }}.wiki`. It should simplify authentication concerns, and is worth checking out.

**Comment by jcbhmr** [comment]
> I just remembered that `gh` is installed in actions, and we can use that. Just set the environment variable `GH_TOKEN` when using `gh`, and clone with `gh repo clone ${{ inputs.repository }}.wiki`. It should simplify authentication concerns, and is worth checking out.

ill have to try that! idk if itll work since it's not cloning a _project_, but cloning the .wiki.git git repo.

basically, the question is: does this work?

```sh
gh repo clone spenserblack/actions-wiki.wiki
gh repo clone spenserblack/actions-wiki.wiki.git
```
???

i havent tried so it might work, or might not 🤷‍♂️

**Comment by jcbhmr** [comment]
also question is if `gh repo clone` supports different non-github.com enterprie-hosted github instances. basically, does it support `$GITHUB_SERVER_URL`

**Comment by jcbhmr** [comment]
@spenserblack 🥳 it works!
image

**Comment by jcbhmr** [comment]
i think merge this with the $GITHUB_ACTOR thing, and then in the future consider the gh cli instead? sound good @spenserblack ?

**Review by spenserblack — APPROVED** [review]

**Pull request merged**